### DISCUSSION TITLE
Csla and Mobile apps

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4134] [Questions]
I have been asked to look into mobile apps (again : )).
How is your experience developing mobile apps (Android and iOS) with Csla?
Could you please let me know what you used? MAUI?

Kind regards

**Comment by kcabral817** [comment]
I'm interested also.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Chicagoan ***@***.***>
Sent: Monday, August 5, 2024 12:20:55 PM
To: MarimerLLC/csla ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [MarimerLLC/csla] Csla and Mobile apps (Discussion #4134)


I have been asked to look into mobile apps (again : )).
How is your experience developing mobile apps (Android and iOS) with Csla?
Could you please let me know what you used? MAUI?

Kind regards

**Comment by rockfordlhotka** [comment]
The technologies I'd look at from the .NET space are:

* MAUI Blazor Hybrid - build your app using Blazor, hosted as a native app in MAUI; pros are that you get Blazor, cons are that it is harder to make the UI different for each platform
* MAUI - build your app using MAUI XAML (successor to Xamarin.Forms); pros are that you get all the MAUI features and libraries, cons are per-platform differences
* Platform.Uno - build your app using a dialect of XAML on a very robust toolset and platform; pros are that Uno is (imo) more capable than MAUI XAML and also supports WebAssembly, cons are that it is a third-party product (open-source with (I think) some paid options)
* Avalonia - build your app using WPF style XAML; pros are the WPF XAML that lots of people know, cons are that it is a third-party product

CSLA _should_ work with all of these options. I _know_ it works with MAUI-Blazor, MAUI, and Uno. I haven't tested with Avalonia.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  Thank you Sir. What's your recommendation among the options listed above?
Kind Regards

**Reply by rockfordlhotka** [reply]
@Chicagoan2016 I'm a consultant, so my answer is always "it depends" 😉

Seriously though, it really does depend on a number of things - your existing team's skill set, what kind of mobile user experience you want to provide, and many other factors.

As a _general_ rule I like the MAUI Hybrid approach, because I like Blazor quite a lot, and that's a good way to build a consistent (not customized) experience across all platforms.

I also really like and admire what the Uno team has done. Their tooling, community and support is quite impressive.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  Could we have a simple example of Hybrid approach or Uno? something that can access, say, camera.

I found link to an Uno app someone on Stackoverflow has shared

https://awc.org/calculators/heights-and-areas-calculator/

**Reply by Chicagoan2016** [reply]
something interesting, are these two separate from MAUI?
https://github.com/xamarin/xamarin-macios

and
https://github.com/dotnet/android

**Reply by rockfordlhotka** [reply]
I suspect those are the two "Xamarin native" technologies that still exist. Not Xamarin.Forms (now MAUI), but rather tailored UI per-platform with C# behind for coding.
